pkg/fqdn/cache.go

```diff
@@ -193,29 +198,31 @@ func (c *DNSCache) Update(lookupTime time.Time, name string, ips []netip.Addr, t
 
 	c.mu.Lock()
 	defer c.mu.Unlock()
+	for _, cache := range updates {
+		c.updated.Insert(cache)
+	}
 	return c.updateWithEntry(entry)
 }
 
 // updateWithEntry implements the insertion of a cacheEntry. It is used by
 // DNSCache.Update and DNSCache.UpdateWithEntry.
-// This needs a write lock
-func (c *DNSCache) updateWithEntry(entry *cacheEntry) bool {
-	changed := false
+// This needs a write lock. Returns true if one or more IPs added to the cache
+// were new.
+// It returns two booleans that indicate if the dns cache was updated, and if one
+// or more IPs were new and therefore upserted
+func (c *DNSCache) updateWithEntry(entry *cacheEntry) (bool, bool) {
 	entries, exists := c.forward[entry.Name]
 	if !exists {
-		changed = true
```

Comment:
I figured out this is a bit confusing, so decided to remove. We already know that if we are adding _something_ here, and the fqdn did not exist in the cache - it will always add one or more entries.